### PR TITLE
Implementation of sheltering transfer

### PR DESCRIPTION
**Pull request by AndrewAR2**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: test/integration/workers/atlas_worker.js [hunk 1]
```diff
@@ -45,17 +47,27 @@ describe('Atlas worker - integration', () => {
     info: () => {},
     error: () => {}
   };
-  let mockStrategy;
+  let mockChallengeStrategy;
+  let mockTransferStrategy;
```
Comment (krzysztof-jelski) [inline test/integration/workers/atlas_worker.js]: Unused variable.

File: test/integration/workers/atlas_worker.js [hunk 2]
```diff
@@ -45,17 +47,27 @@ describe('Atlas worker - integration', () => {
...
-      shouldResolveChallenge: sinon.stub().resolves(true),
-      afterChallengeResolution: sinon.stub()
+      shouldResolve: sinon.stub().resolves(true),
+      afterResolution: sinon.stub()
     };
-    mockStrategy.__proto__ = AtlasChallengeParticipationStrategy.prototype;
+    mockChallengeStrategy.__proto__ = AtlasParticipationStrategy.prototype;
+  };
+
+  const createMockTransferStrategy = () => {
```
Comment (krzysztof-jelski) [inline test/integration/workers/atlas_worker.js]: This is not needed, as `mockTransferStrategy` is not used in the Atlas worker.

However, I suggest creating an integration test similar to this one for transfers. Either in this file or in separate one.

File: src/workers/atlas_resolvers/atlas_transfer_resolver.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  async tryToResolve({bundleId}, {transferId}) {
+    await this.transfersRepository.resolveTransfer(transferId);
+    await this.dataModelEngine.markBundleAsSheltered(bundleId);
+    await this.addLog('🍾 Yahoo! The bundle is ours.', {bundleId});
+  }
+
+  async tryToDownload({donorId, bundleId, transferId}) {
+    const transferExpirationTime = await this.transfersRepository.getTransferExpirationTimeInMs(transferId);
```
Comment (krzysztof-jelski) [inline src/workers/atlas_resolvers/atlas_transfer_resolver.js]: The method `getTransferExpirationTimeInMs` does not exist in `transfersRepository`.

File: src/services/transfers_repository.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      ['transferId']);
+    this.activeResolutionsCache.applyIncomingResolutionEvents(startedTransfers, resolvedTransfers, cancelledTransfers);
+  }
+
+  async getFromBlock() {
+    if (this.lastSavedBlock > 0) {
+      return this.lastSavedBlock + 1;
+    }
+    const transferDuration = await this.configWrapper.challengeDuration();
+    return this.shelteringTransfersWrapper.earliestMeaningfulBlock(transferDuration);
```
Comment (krzysztof-jelski) [inline src/services/transfers_repository.js]: The method `earliestMeaningfulBlock` does not exist in `shelteringTransfersWrapper`.

Comment (AndrewAR2) [reply src/services/transfers_repository.js]: > Also there are tons of duplication between transfer and challenges classes.

For example?

File: test/services/transfers_repository.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        {transferId, blockNumber: 2, logIndex: 1}
+      ]);
+    });
+  });
+
+  describe('ongoingTransfers', () => {
+    const donorId = 1;
+    const bundleId = 2;
+    const transferId = 3;
+    // const fromBlock = 4;
```
Comment (Anze1m) [inline test/services/transfers_repository.js]: Comment should be removed

Comment (AndrewAR2) [reply test/services/transfers_repository.js]: This is a temporary solution - the code has changed and it is necessary to rewrite the test code.

Comment (AndrewAR2) [reply test/services/transfers_repository.js]: Fixed
